src/core/login/pages/reconnect/reconnect.ts

```diff
@@ -185,49 +188,60 @@ export class CoreLoginReconnectPage {
         }
 
         if (!this.appProvider.isOnline()) {
-            this.domUtils.showErrorModal('core.networkerrormsg', true);
-
-            return;
-        }
-
-        const modal = this.domUtils.showModalLoading();
-
-        // Start the authentication process.
-        this.sitesProvider.getUserToken(siteUrl, username, password).then((data) => {
-            return this.sitesProvider.updateSiteToken(this.infoSiteUrl, username, data.token, data.privateToken).then(() => {
-
-                this.domUtils.triggerFormSubmittedEvent(this.formElement, true);
-
-                // Update site info too because functions might have changed (e.g. unisntall local_mobile).
-                return this.sitesProvider.updateSiteInfoByUrl(this.infoSiteUrl, username).then(() => {
-                    // Reset fields so the data is not in the view anymore.
-                    this.credForm.controls['password'].reset();
-
-                    // Go to the site initial page.
-                    return this.loginHelper.goToSiteInitialPage(this.navCtrl, this.pageName, this.pageParams);
-                }).catch((error) => {
-                    if (error.loggedout) {
-                        this.loginHelper.treatUserTokenError(siteUrl, error, username, password);
-                    } else {
-                        this.domUtils.showErrorModalDefault(error, 'core.login.errorupdatesite', true);
-                    }
+            this.validateCredentialsOffline(username, password).then((success) => {
+                if (success) {
```

Comment:
amazing how bad the diff is. I think enterprise bitbucket puts this to shame. It looks like just this not online block has changed

Reply:
actually this is confusing because the !online check already existed. confused me for a sec too - i thought i didn't push a line lol